File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const hallOfFame = [];
    const lowestArr = [];
    let lowest = 0;
    
    for (let i = 0; i < score.length; i++) {
        if (i < k) { // i가 k번째보다 작을 경우
            hallOfFame.push(score[i]); // 명예의 전당에 넣기
            hallOfFame.sort((a, b) => a - b); // 정렬
            lowest = hallOfFame[0] // 최저값 0번째
            lowestArr.push(lowest); // 
        } else { // i > k번째
            if (score[i] > lowest) {
                hallOfFame.shift(); // k번째 방출
                hallOfFame.push(score[i]); // score[i] 넣기
                hallOfFame.sort((a, b) => a - b); // 정렬
            }
            lowest = hallOfFame[0] // 최저값 0번째
            lowestArr.push(lowest); //
        }
    }
    
    return lowestArr;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    
    hallOfFame = []
    for s in score:
        if len(hallOfFame) < k:
            hallOfFame.append(s) # [10, 100, 20]
        elif s > hallOfFame[-1]:
            hallOfFame.pop()
            hallOfFame.append(s)
        hallOfFame.sort(reverse=True)
        answer.append(hallOfFame[-1])
    
    return answer